src/arch/riscv/pcstate.hh

```diff
@@ -43,29 +43,39 @@
 #define __ARCH_RISCV_PCSTATE_HH__
 
 #include "arch/generic/pcstate.hh"
+#include "arch/riscv/regs/vec.hh"
+#include "base/bitunion.hh"
 
 namespace gem5
 {
 
 namespace RiscvISA
 {
 
-class PCState : public GenericISA::UPCState<4>
+BitUnion32(VTYPE)
+    Bitfield<31> vill;
+    Bitfield<7> vma;
+    Bitfield<6> vta;
+    Bitfield<5, 3> vsew;
+    Bitfield<2, 0> vlmul;
+EndBitUnion(VTYPE)
+
+class PCState : public GenericISA::UPCState<16>
 {
   private:
     bool _compressed = false;
     bool _rv32 = false;
 
   public:
-    using GenericISA::UPCState<4>::UPCState;
+    using GenericISA::UPCState<16>::UPCState;
```

Comment:
This should not be changed to 16. UPCState<4> means the InstructionWidth of the micro op is 4-byte long.